pkg/controller/servicediscovery/httpnamespace/setup.go

```diff
@@ -72,3 +75,15 @@ func postCreate(_ context.Context, cr *svcapitypes.HTTPNamespace, resp *svcsdk.C
 	cr.SetOperationID(resp.OperationId)
 	return cre, err
 }
+
+func preUpdate(_ context.Context, cr *svcapitypes.HTTPNamespace, obj *svcsdk.UpdateHttpNamespaceInput) error {
+	if meta.GetExternalName(cr) != "" {
+		obj.Id = awsclient.String(meta.GetExternalName(cr))
+		obj.Namespace = &svcsdk.HttpNamespaceChange{
```

Comment:
Can we target a specific field in case they decide to expand `svcsdk.HttpNamespaceChange` struct and the code-gen sets some fields, hence we end up overriding them?

Reply:
Goes for all resources in `servicediscovery`